src/components/DetailsMenu/CLIInstructions/CliInstructions.test.tsx

```diff
@@ -0,0 +1,108 @@
+import { renderComponentWithHook } from "test_utils/TestHooks";
+import { mockEnvironmentVariables } from "test_utils/utils";
+import CLIInstructions from ".";
+
+const { mockEnv, cleanup } = mockEnvironmentVariables();
+
+const wrapper = ({ children }: { children: React.ReactNode }) => (
+  <LogContextProvider initialLogLines={[]}>{children}</LogContextProvider>
+);
+
+describe("cliInstructions", () => {
+  afterEach(() => {
+    cleanup();
+  });
+  it("should not exist if no task id or build id is provided", () => {
+    const { Component } = renderComponentWithHook(
+      useLogContext,
+      <CLIInstructions />
+    );
+    render(<Component />, { wrapper });
+    expect(screen.queryByDataCy("cli-instructions")).not.toBeInTheDocument();
+  });
+  it("should render the task log download command with the appropriate tags if the log is a task log", () => {
+    const { Component, hook } = renderComponentWithHook(
+      useLogContext,
+      <CLIInstructions />
+    );
+    render(<Component />, { wrapper });
+    act(() => {
+      hook.current.setLogMetadata({
+        taskID: "taskId",
+        execution: "0",
+        origin: "task",
+        logType: LogTypes.EVERGREEN_TASK_LOGS,
+      });
+    });
+    let taskDownloadCommand =
+      "evergreen buildlogger fetch --task_id taskId --execution 0 --tags task_log";
+    expect(screen.getByText(taskDownloadCommand)).toBeInTheDocument();
+    act(() => {
+      hook.current.setLogMetadata({
+        taskID: "taskId",
+        execution: "0",
+        origin: "agent",
+        logType: LogTypes.EVERGREEN_TASK_LOGS,
+      });
+    });
+    taskDownloadCommand =
+      "evergreen buildlogger fetch --task_id taskId --execution 0 --tags agent_log";
+    expect(screen.getByText(taskDownloadCommand)).toBeInTheDocument();
+    act(() => {
+      hook.current.setLogMetadata({
+        taskID: "taskId",
+        execution: "0",
+        origin: "all",
+        logType: LogTypes.EVERGREEN_TASK_LOGS,
+      });
+    });
+    taskDownloadCommand =
+      "evergreen buildlogger fetch --task_id taskId --execution 0";
+    expect(screen.getByText(taskDownloadCommand)).toBeInTheDocument();
+  });
+  it("should render the test log download command if the log is a test log", () => {
+    const { Component, hook } = renderComponentWithHook(
+      useLogContext,
+      <CLIInstructions />
+    );
+    render(<Component />, { wrapper });
+    act(() => {
+      hook.current.setLogMetadata({
+        taskID: "taskId",
+        execution: "0",
+        testID: "testId",
+        logType: LogTypes.EVERGREEN_TEST_LOGS,
+      });
+    });
+    const testDownloadCommand =
+      "evergreen buildlogger fetch --task_id taskId --execution 0 --test_name testId";
+    expect(screen.getByText(testDownloadCommand)).toBeInTheDocument();
+  });
+  it("should render the logkeeper download command if the log is a resmoke log", () => {
+    mockEnv("REACT_APP_LOGKEEPER_URL", "https://logkeeper-test.org");
```

Comment:
I think this doesn't actually do anything; the URL seems to still be `undefined` if you print `logkeeperDownloadCommand`